lib/Fullness/Fullness.cpp

```diff
@@ -0,0 +1,99 @@
+/**
+ * @file Fullness.cpp
+ * @brief C file for Fullness class
+ * @version 0.1
+ * @date 2022-05-04
+ *
+ */
+
+#include "Fullness.hpp"
+#include "IDistance.hpp"
+#include <Arduino.h>
+#include <bits/stdc++.h>
+#include <cstdint>
+
+using namespace Zotbins;
+
+#define SENSOR_CALLS 8 // made it a number divisible by 4 for IQM calculations
+
+// using constructor
+Fullness::Fullness(int32_t binHeight, IDistance::IDistance &distanceSensor)
+{
+    height = binHeight;
+    sensor = &distanceSensor;
+}
+
+float Fullness::getFullness()
+{
+    // call getDistance on sensor SENSOR_CALLS times to get an array of distances
+    int distances[SENSOR_CALLS] = {};
+    for (int i = 0; i < SENSOR_CALLS; i++)
+    {
+        if (sensor->getDistance() >= 0) // checking getDistance to make sure function won't get stuck
+        {
+            distances[i] = sensor->getDistance();
```

Comment:
Don't error check here because if we get 7 values readings and 1 bad reading, it is still fine